src/main.cpp

```diff
@@ -7626,70 +7626,74 @@ bool SendMessages(CNode *pto)
         //
         // Message: inventory
         //
-        std::vector<CInv> vInv;
+
         std::vector<CInv> vInvWait;
+        std::vector<CInv> vInvSend;
         {
-            bool fSendTrickle = pto->fWhitelisted;
+            bool fSendTrickle = !pto->fWhitelisted;
             if (pto->nNextInvSend < nNow)
             {
-                fSendTrickle = true;
+                fSendTrickle = false;
                 pto->nNextInvSend = PoissonNextSend(nNow, AVG_INVENTORY_BROADCAST_INTERVAL);
             }
-            LOCK(pto->cs_inventory);
-            vInv.reserve(std::min<size_t>(1000, pto->vInventoryToSend.size()));
-            vInvWait.reserve(pto->vInventoryToSend.size());
-            BOOST_FOREACH (const CInv &inv, pto->vInventoryToSend)
+
+            if (1)
             {
-                if (inv.type == MSG_TX && pto->filterInventoryKnown.contains(inv.hash))
-                    continue;
+                // BU - here we only want to forward message inventory if our peer has actually been requesting
+                // useful data or giving us useful data.  We give them 2 minutes to be useful but then choke off
+                // their inventory.  This prevents fake peers from connecting and listening to our inventory
+                // while providing no value to the network.
+                // However we will still send them block inventory in the case they are a pruned node or wallet
```

Comment:
AFAIK pruned node send headers, forward (tips) blocks and transactions,  what they don't do is serving historical blocks for IBD.

Reply:
this comment isn't changed from the previous version.  It just got shifted.  I'm not quite sure what you are saying but if we need to change the comment, let's do it in a new PR.